x-pack/plugins/ingest_manager/server/plugin.ts

```diff
@@ -112,7 +114,11 @@ const allSavedObjectTypes = [
  */
 export type ExternalCallback = [
   'packagePolicyCreate',
-  (newPackagePolicy: NewPackagePolicy) => Promise<NewPackagePolicy>
+  (
+    newPackagePolicy: NewPackagePolicy,
+    context: RequestHandlerContext,
```

Comment:
AFAIK no other users of the callback yet, so we don't know their needs. security solution definitely needs these (at present) so they will need to be provided. But if other plugins registering and receiving the callback are free to ignore these arguments. I believe marking these optional would do two things: 1) make a different callback-receiver write their function signature to be shorter, omitting the optional params (I think?) instead of including them as arguments and not using them. And 2) technically allow for ingest to leave off providing these arguments.

Since security solution requires them, I want to avoid for the possibility of effect number 2